docs/spec/spec.md

```diff
@@ -0,0 +1,1294 @@
+   <table >
+<tr>
+<th> Ballerina </th>
+<th> Proto message </th>
+ </tr>
+
+<tr></tr>
+
+<tr>
+<td>
+<pre>
+type Student record {
+    string name;
+    int courseId;
+    decimal fees;
+};
+</pre>
+<pre>
+type Teacher record {
+    string name;
+    int courseId;
+    decimal salary;
+};
+</pre>
+<pre>
+type <b>TupleWithRecord</b> [Student, Teacher];
+</pre>
+</td>
+<td>
+<pre>
+message <b>TupleBuilder</b> {
+  message Teacher {
+    message DecimalValue {
+       uint32 scale  = 1;
+       uint32 precision  = 2;
+       bytes value  = 3;
+    }
+    sint64 courseId  = 1;
+    string name  = 2;
+    DecimalValue salary  = 3;
+  }
+  message Student {
+    message DecimalValue {
+      uint32 scale  = 1;
+      uint32 precision  = 2;
+      bytes value  = 3;
+    }
+    sint64 courseId  = 1;
+    DecimalValue fees  = 2;
+    string name  = 3;
+  }
+  Student element_1  = 1;
+  Teacher element_2  = 2;
+}
+</pre>
+</td>
+</tr>
+<table>
+
+ 5. Tuple with tuple elements
+    <table >
+<tr>
+<th> Ballerina </th>
+<th> Proto message </th>
+ </tr>
+
+<tr></tr>
+
+<tr>
+<td>
+<pre>
+type PrimitiveTuple [byte, int, float, boolean, string ,decimal];
+</pre>
+<pre>
+type TupleWithUnion [byte|string, decimal|boolean];
+</pre>
+<pre>
+type <b>TupleOfTuples</b> [PrimitiveTuple, TupleWithUnion];
+</pre>
+</td>
+<td>
+<pre>
+message TupleBuilder {
+  message element_2___TupleBuilder {
+    message element_1___UnionBuilder {
+       bytes byte___unionField  = 1;
+       string string___unionField  = 2;
+    }
+    message element_2___UnionBuilder {
+     message DecimalValue {
+        uint32 scale  = 1;
+        uint32 precision  = 2;
+        bytes value  = 3;
+      }
+      bool boolean___unionField  = 1;
+      DecimalValue decimal___unionField  = 2;
+    }
+    element_1___UnionBuilder element_1  = 1;
+    element_2___UnionBuilder element_2  = 2;
+  }
+  message element_1___TupleBuilder {
+    message DecimalValue {
+       uint32 scale  = 1;
+       uint32 precision  = 2;
+       bytes value  = 3;
+    }
+    bytes element_1  = 1;
+    sint64 element_2  = 2;
+    double element_3  = 3;
+    bool element_4  = 4;
+    string element_5  = 5;
+    DecimalValue element_6  = 6;
+  }
+  element_1___TupleBuilder element_1  = 1;
+  element_2___TupleBuilder element_2  = 2;
+}
+</pre>
+</td>
+</tr>
+<table>
+
+### 3.8 Enum
+Ballerina enum is a syntactic sugar of union of constant strings thus enum is handled as union in protobuf level
+     <table >
+<tr>
+<th> Ballerina </th>
+<th> Proto message </th>
+ </tr>
+
+<tr></tr>
+
+<tr>
+<td>
+<pre>
+enum Color {
+    RED=”red”,
+    GREEN,
+    BLUE
+}
+</pre>
+</td>
+<td>
+<pre>
+message UnionBuilder {
+   string string___unionField  = 1;
+}
+</pre>
+</td>
+</tr>
+<tr></tr>
+
+<tr>
+<td>
+<pre>
+const OPEN = "open";
+const CLOSE = "close";
+type <b>STATE</b> OPEN|CLOSE;
+</pre>
+</td>
+<td>
+<pre>
+message <b>UnionBuilder</b> {
+   string string___unionField  = 1;
+}
+</pre>
+</td>
+</tr>
+<table>
+
+
+
+## 4. Sample
```

Comment:
Let's add samples under each section when needed. We don't need a complete sample as a separate section.

Reply:
+1, moved to appropriate sections